src/client/components/pages/Courses/CourseInstanceTableBody.tsx

```diff
@@ -0,0 +1,99 @@
+import { CellLayout } from 'client/components/general';
+import { COURSE_TABLE_COLUMN, getAreaColor, TERM } from 'common/constants';
+import CourseInstanceResponseDTO from 'common/dto/courses/CourseInstanceResponse';
+import {
+  TableBody, TableRow, TableRowHeadingCell, VALIGN, TableCell,
+} from 'mark-one';
+import React, { FunctionComponent, ReactElement, Ref } from 'react';
+import { CourseInstanceListColumn } from './tableFields';
+
+interface CourseInstanceTableBodyProps {
+  /**
+   * The list of courses to be shown in the table
+   */
+  courseList: CourseInstanceResponseDTO[];
+  /**
+   * The data to display
+   */
+  tableData: CourseInstanceListColumn[];
+  /**
+   * Controls the opening of the meeting modal with the requested course and term
+   */
+  openMeetingModal: (course: CourseInstanceResponseDTO, term: TERM) => void;
+  /**
+   * Controls the opening of the instructor modal with the requested course and term
+   */
+  openInstructorModal: (course: CourseInstanceResponseDTO, term: TERM) => void;
```

Comment:
I'd maybe extract these two functions as a separate type..


```ts
type OpenModalCallback = (course: CourseInstanceResponseDTO, term: TERM) => void;

// ..

/**
 * Controls the opening of the meeting modal with the requested course and term
 */
openMeetingModal: OpenModalCallback;
/**
 * Controls the opening of the instructor modal with the requested course and term
 */
openInstructorModal: OpenModalCallback;
```

That makes it a little clearer  that the two functions have the same type signature. It also allows you to provide additional doc blocks for the actual function (should you wish to do so)

Reply:
Thanks for suggesting this! This was added in [e9da57b](https://github.com/seas-computing/course-planner/pull/456/commits/e9da57bc5a736e749fbfe2e2c6c0370a41d04bde).